src/spec/nrepl/spec.clj

```diff
@@ -0,0 +1,104 @@
+(ns nrepl.spec
+  "This serves as a reference for messages as implemented by the default nrepl
+  middlewares. This is not a formal specification."
+  (:require [clojure.spec.alpha :as s]
+            [clojure.string :as str]))
+
+;; General
+
+(s/def ::uuid-str #(and (string? %)
+                        (re-matches #"[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}"
+                                    (str/lower-case %))))
+
+;; should be uuid strings
+(s/def ::session (s/or :single ::uuid-str
+                       :multi  (s/coll-of ::uuid-str :kind? set?)))
+
+;; Clone
```

Comment:
Might be a good idea to have a separate section for the request and response keys for every op. That'd be easier to read IMO.

Reply:
Do you mean we should organise this source file better? If so, I've tried to roughly do that, but some keys are used in more than one op.

Reply:
I was thinking we can start with a section "Common" for things like `id`, `session`, etc. 

`eval` and `load-file` can also be grouped together.

Reply:
Can do.